src/renderer/src/routes/(setup)/quick-setup-step-2/+page.svelte

```diff
@@ -17,11 +18,36 @@
 
 	const quickSetup = client.quickSetup.createMutation();
 	const launch = client.launch.createMutation();
+	const lairSetupRequired = client.lairSetupRequired.createQuery();
 
 	let confirmPasswordInput = '';
 
 	let launching = false;
 
+	let cannotGoBack = window.sessionStorage.getItem('quickSetupFailedHalfway');
+
+	/**
+	 * Check at which step launching failed and route accordingly.
+	 * If it failed after keystore initialization go to the enter-password ('unlock') screen,
+	 * otherwise stay on this screen.
+	 */
+	const handleErrorAndDetermineRoute = async (error: unknown) => {
+		launching = false;
+		setupProgress.set('');
+		console.error(error);
+		toastStore.trigger({
+			message: $i18n.t((error as Error).message || 'unknownError'),
+			background: 'variant-filled-error'
+		});
+		const required = await $lairSetupRequired.refetch();
+		if (required.isSuccess && required.data === false) {
+			const persistedStore = new LauncherPersistedStore();
+			await persistedStore.quickSetupChosen.set(true);
+			window.sessionStorage.setItem('quickSetupFailedHalfway', 'true');
```

Comment:
Might need to ensure that this session storage item is cleared when it's no longer needed, possibly in the signupAndLaunch function if the setup completes successfully?